From bdd14d77f93829b38574af88110f0e079eec0d40 Mon Sep 17 00:00:00 2001
From: William Wu <william.wu@rock-chips.com>
Date: Fri, 20 Nov 2020 15:42:23 +0800
Subject: [PATCH] phy: rockchip: inno-usb2: fix suspend control for rk3568 phy0
 host port

RK3568 USB2.0 PHY0 Host port is used for USB3.0 Host interface.
When USB3.0 Host controller is working at super speed, it may
still need the USB2.0 PHY0 to work in normal mode and get clks
from the PHY0. So we select PHY suspend control from USB3.O
Host controller utmi interface.

Change-Id: I66df057e200d04f66082257d26120764c54af3cf
Signed-off-by: William Wu <william.wu@rock-chips.com>
---
 drivers/phy/rockchip/phy-rockchip-inno-usb2.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/phy/rockchip/phy-rockchip-inno-usb2.c b/drivers/phy/rockchip/phy-rockchip-inno-usb2.c
index e140abdb07ce..f49d8f9c7b75 100644
--- a/drivers/phy/rockchip/phy-rockchip-inno-usb2.c
+++ b/drivers/phy/rockchip/phy-rockchip-inno-usb2.c
@@ -2833,7 +2833,8 @@ static const struct rockchip_usb2phy_cfg rk3568_phy_cfgs[] = {
 				.utmi_ls	= { 0x00c0, 5, 4, 0, 1 },
 			},
 			[USB2PHY_PORT_HOST] = {
-				.phy_sus	= { 0x0004, 8, 0, 0x1d2, 0x1d1 },
+				/* Select suspend control from controller */
+				.phy_sus	= { 0x0004, 8, 0, 0x1d2, 0x1d2 },
 				.ls_det_en	= { 0x0080, 1, 1, 0, 1 },
 				.ls_det_st	= { 0x0084, 1, 1, 0, 1 },
 				.ls_det_clr	= { 0x0088, 1, 1, 0, 1 },
-- 
2.35.3

